README.md

```diff
@@ -1 +1,186 @@
+
+```js
+const applyProgressBeforeInteractive = (elements, progress) => {
+    // elements.foo
+    // elements.bar
+};
+
+const MyPage = () => (
+    <WaitForReact
+        applyProgressBeforeInteractive={ applyProgressBeforeInteractive }
+        promise={ promise }>
+        { ({ progress }) => (
+            <div className={ classNames(styles.progressBarWrapper, progress > 1 && styles.done) }>
+                <div
+                    data-wait-for-react-element="foo"
+                    className={ styles.progressBarHorizontal }
+                    style={ { transform: `scaleX(${progress})` } } />
+                <div
+                    data-wait-for-react-element="bar"
+                    className={ styles.progressBarVertical }
+                    style={ { transform: `scaleY(${progress})` } } />
+            </div>
+        ) }
+    </WaitForReact>
+);
+```
+
+### progressDecay
+
+Type: `string`   
+Default: `function (time) { return 1 - Math.exp(-1 * time / 4000); }`
+
+A function in it's string form to calculate the progress value based on the elapsed time. Typically, the algorithm has a decay pattern, where increments are smaller and smaller as time passes by.
+
+`<WaitForReact>` will call `progressDecay` to simulate a "fake progress" until your app becomes interactive. To make this possible, `progressDecay` will be added in an inline script included as part SSR or static export. Similarly, `<WaitForReact>` will call `progressDecay` to simulate a "fake progress" if a standard promise is passed as the `promise` prop.
+
+⚠️ The reason for this prop to be a string instead of a function has to do with compilation. Because server-side compilation usually differ from client-side compilation, the actual function in it's string form would be different and React would complain with a mismatch warning when rehydrating. Having that said, you should be careful in how you write this function so that it's compatible with all your target environments.
+
+The `progressDecay` function signature is `(time) => <progress>`, where `time` is the elapsed time in milliseconds. It must return the `progress` value in the form of a number between 0 and 1 (exclusive).
+
+### progressInterval
+
+Type: `number`   
+Default: 100
+
+The interval, in ms, to report progress. The value of `progressInterval` will effectively throttle all the internal behavior of `<WaitForReact>`, including the frequency in which the `children` render prop will be called.
+
+ℹ️ If you are using CSS transitions, the transition durations should be equal or smaller than `progressInterval`. This circumvents an issue in Chrome, Safari and other WebKit/Blink based browsers where updating a CSS property in the middle of a transition will cause the animation to "restart".
```

Comment:
I think Chrome doesn't have this issue, but Firefox does :+1:

Reply:
Chrome has the issue too unfortunately. I spotted it while developing the demo.

Reply:
I've rephrased this a bit to add Firefox as well.